torch/utils/_pytree.py

```diff
@@ -26,194 +28,215 @@
 To improve the performance we can move parts of the implementation to C++.
 """
 
+"""
+TreeSpecSchema is the schema used to serialize the TreeSpec
+It contains the following fields:
+- type: A string name of the type. null for the case of a LeafSpec.
+- context: A string format of the context, optional.
```

Comment:
Why is the context always a string format? It seems like it would conveniently be other JSON types sometimes.

Reply:
There's 2 types of dataclasses: one is the TreeSpec which is what pytree.flatten returns, and I added a TreeSpecSchema, which is what we serialize to at the end. So I was thinking that all the fields strings -- by default we serialize the context to string using json.dumps, and users can plug in their own serializer for converting some custom data structure to string. I guess my thinking is to be more flexible with the context -- it can be non-json serializable if users plug in their own serializer?